identity-service/src/routes/fp.js

```diff
@@ -0,0 +1,82 @@
+const {
+  handleResponse, successResponse, errorResponseBadRequest, errorResponse
+} = require('../apiHelpers')
+const { QueryTypes } = require('sequelize')
+
+const models = require('../models')
+
+const { logger } = require('../logging')
+
+const getDeviceIDCountForUserId = async (userId) => {
+  // Get the # of users sharing any visitorID
+  // associated with `userId` on any platform
+  const res = await models.sequelize.query(
+    `select count(*), "Fingerprints"."userId"
+     from "Fingerprints"
+     where "visitorId" in (
+      select distinct "visitorId"
+      from "Fingerprints"
+      where "userId" = :userId
+    ) group by "Fingerprints"."userId"`,
+    {
+      replacements: {
+        userId
+      },
+      type: QueryTypes.SELECT
+    }
+  )
+  return res.length
```

Comment:
since you just return res.length here, you could probably avoid the count right and just do a group by?

```
select "Fingerprints"."userId"
     from "Fingerprints"
     where "visitorId" in (
      select distinct "visitorId"
      from "Fingerprints"
      where "userId" = :userId
    ) group by "Fingerprints"."userId"`,
```

or even just `select 1` i guess, but that's less legible

Reply:
yep nice